### DISCUSSION TITLE
Network card modeling for servers

### DISCUSSION THREAD

**Discussion by ciroiriarte** [netbox-community/devicetype-library#1342] [Q&A] [answered]
Hello!,

What would be the correct way to model Ethernet NIC or Fiber channel HBAs?.

I would like to:
- model *Base-T cards (copper) with multiple fixed ports.
- model NIC cards with multiple ports which can host transceivers (applies the same to FC HBAs)
- track transceiver vs port

I've seen public examples that use devices for a NIC, others use modules.

What would be the correct approach?

**Comment by danner26** [comment, marked as answer]
Since a nic card is not a device itself, the best way to model it would be in the module-types

These modules would then slot into a module-bay within a device
